pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -310,6 +311,63 @@ func newOpenStack(cfg Config) (*OpenStack, error) {
 	return &os, nil
 }
 
+func configFromEnv() (cfg Config, ok bool) {
+	cfg.Global.AuthUrl = os.Getenv("OS_AUTH_URL")
+
+	cfg.Global.TenantId = os.Getenv("OS_TENANT_ID")
+	// Rax/nova _insists_ that we don't specify both tenant ID and name
+	if cfg.Global.TenantId == "" {
+		cfg.Global.TenantName = os.Getenv("OS_TENANT_NAME")
+	}
+
+	cfg.Global.Username = os.Getenv("OS_USERNAME")
+	cfg.Global.Password = os.Getenv("OS_PASSWORD")
+	cfg.Global.Region = os.Getenv("OS_REGION_NAME")
+
+	cfg.Global.TenantName = os.Getenv("OS_TENANT_NAME")
+	if cfg.Global.TenantName == "" {
+		cfg.Global.TenantName = os.Getenv("OS_PROJECT_NAME")
+	}
+
+	cfg.Global.TenantId = os.Getenv("OS_TENANT_ID")
+	if cfg.Global.TenantId == "" {
+		cfg.Global.TenantId = os.Getenv("OS_PROJECT_ID")
+	}
+
+	cfg.Global.DomainId = os.Getenv("OS_DOMAIN_ID")
+	if cfg.Global.DomainId == "" {
+		cfg.Global.DomainId = os.Getenv("OS_USER_DOMAIN_ID")
+	}
+
+	cfg.Global.DomainName = os.Getenv("OS_DOMAIN_NAME")
+	if cfg.Global.DomainName == "" {
+		cfg.Global.DomainName = os.Getenv("OS_USER_DOMAIN_NAME")
+	}
+
+	ok = (cfg.Global.AuthUrl != "" &&
+		cfg.Global.Username != "" &&
+		cfg.Global.Password != "" &&
+		(cfg.Global.TenantId != "" || cfg.Global.TenantName != "" ||
+			cfg.Global.DomainId != "" || cfg.Global.DomainName != ""))
+
+	cfg.Metadata.SearchOrder = fmt.Sprintf("%s,%s", configDriveID, metadataID)
+	cfg.BlockStorage.BSVersion = "auto"
+
+	return
+}
+
+func GetOpenstack() (*OpenStack, error) {
+
+	cfg, ok := configFromEnv()
+	if !ok {
+		fmt.Printf("No config found in environment")
```

Comment:
use `fmt.Errorf` which is more common in the e2e tests.
Also the typical e2e error msg format: begins lower case, use %q for longer inserts, use %v for the actual err insert (which is almost always at the end of the msg proceded with a ":"), and format the msg as: `fmt.Errorf("msg-1 %q msg-2: %v, foo-1, err)`

Reply:
We keep calling logic from the pkg/cloudproviders/providers/openstack  package in this case. This file is not in the e2e/storage package. Should we really use the rules common in the e2e tests in this file.

Reply:
No. Sorry, my mistake.

Reply:
Yes I intended to call newOpenStack so I created a wrapper method GetOpenstack because newOpenStack is unexported (since it is named beginning with lowercase) so I cant call it when am outside its package.